File: backend/src/models/Playlists.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface Iplaylists extends Document {
    name: string,
    creator: string,
    description: string,
    coverImage: string,
    playCount: number,
    path: string,
    updateTime: Date,
    collects: number
    deleted: boolean
}

const PlaylistsSchema: Schema = new Schema({
    name: {
        type: String,
        required: [true, '播放列表名字不能为空'],
        trim: true
    },
    creator: {
        type: String,
        required: [true, '创建者不能为空'],
        trim: true
    },
    description: {
        type: String,
        required: [true, '播放列表描述不能为空'],
        trim: true
    },
    coverImage: {
        type: String,
        required: [true, '封面图片不能为空'],
        trim: true
    },
    playCount: {
        type: Number,
        default: 0
    },
    path: {
        type: String,
        required: [true, '路由路径不能为空'],
        trim: true
    },
    updateTime: {
        type: Date,
        default: Date.now
    },
    collects: {
        type: Number,
        default: 0
    },
    deleted: {
        type: Boolean,
        default: false
    }
}, {
    timestamps: true
})

const Playlists = mongoose.model<Iplaylists>('Playlists', PlaylistsSchema);
export default Playlists;

File: backend/java-backend/src/main/java/com/example/demo/model/mongo/Playlists.java
package com.example.demo.model.mongo;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "playlists")
public class Playlists {

    @Id
    private String id;

    private String name;
    private String creator;
    private String creatorAvatar;
    private String description;
    private String coverImage;
    private Integer playCount;
    private String path;
    private LocalDateTime updateTime;
    private Integer collects;
    private Boolean deleted;
    private LocalDateTime createdAt;
}
